### PR TITLE
Add "Upload Data" page

### PR DESCRIPTION
**Pull request by jsteuver**
This is a page where a user can upload .csv files for households, products, and transactions to be added to the database.

The server expects the uploaded files to follow the same format as the ones found in the sample dataset (i.e. CSV file, column headers, and same column order). *The biggest concern is the static column order; I will reach out to the professor and ask him if this is acceptable or if we should allow for any order of columns.* If we need to allow for any order of columns, we need to rework how attributes are assigned when reading from the file (likely just reading the column headers first and then following the same index for the values).

The products and households uploads will update a row if the corresponding `product_num` or `hshd_num` (respectively) already exists, and only insert if it doesn't exist. This is to allow for things like price changes and family size changes. Transactions, on the other hand, will only insert and never update. Once a transaction in the store has been made, it doesn't really make sense to be modifying the data for it years later.

The best way I have found to test this is to take the sample CSV files that were given, make copies of them, and shorten those copies to only include a handful of rows. You can then modify the values of those rows however you see fit. Just remember to delete any new inserts after you are done testing, as we don't want any junk test data to stay in there longer than it needs to.

If, by chance, any uploaded data causes the server to not work as expected or produces errors, I have made backup tables for each of households, products, and transactions. This will allow us to recover much quicker than trying to import the sample CSV files into the database again.

### CODE REVIEW

**Comment by segeeslice** [comment]
@jsteuver merge conflicts! Sorry I'm good at throwing those at you haha

File: cc_midterm/cc_midterm/forms.py [hunk 1]
```diff
@@ -2,9 +2,16 @@
 from django.contrib.auth.models import User
 from django.contrib.auth.forms import UserCreationForm
 
+from .models import *
```
Comment (segeeslice) [inline cc_midterm/cc_midterm/forms.py]: To my understanding, the models are not used in this file right? Could probably remove this import

File: cc_midterm/cc_midterm/uploads.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                hshd_num=int(row_values[0].strip()),
+                l=row_values[1],
+                age_range=row_values[2],
+                marital=row_values[3],
+                income_range=row_values[4],
+                homeowner=row_values[5],
+                hshd_composition=row_values[6],
+                hh_size=row_values[7],
+                children=row_values[8]
+            )
```
Comment (segeeslice) [inline cc_midterm/cc_midterm/uploads.py]: Really not a huge fan of this, considering there are pretty easy ways around it if you're not familiar, usually via `csv` or `pandas` (although `pandas` is an external lib). For example from [this stack overflow](https://stackoverflow.com/questions/21572175/convert-csv-file-to-list-of-dictionaries):

> Use `csv.DictReader`:
> ```
> import csv
> 
> with open('test.csv') as f:
>     a = [{k: int(v) for k, v in row.items()}
>         for row in csv.DictReader(f, skipinitialspace=True)]
> ```
> Will result in :
> ```
> [{'col2': 2, 'col3': 3, 'col1': 1}, {'col2': 5, 'col3': 6, 'col1': 4}]
> ```
>

Comment (segeeslice) [reply cc_midterm/cc_midterm/uploads.py]: Same goes for other related areas

**Review by segeeslice — CHANGES_REQUESTED** [review]
Looks good overall! But I don't like the static headers as you mentioned yourself, but it's not too difficult to read CSV's as lists of dicts so we wouldn't need to worry about it. I left a comment showing an example of a solution.

Going to request that as a change because I think it's a fairly easy and worthwhile one.

File: cc_midterm/cc_midterm/forms.py (hunk 1 above)
Comment (jsteuver) [reply cc_midterm/cc_midterm/forms.py]: You're right, I went ahead and removed that.

File: cc_midterm/cc_midterm/uploads.py (hunk 2 above)
Comment (jsteuver) [reply cc_midterm/cc_midterm/uploads.py]: Fixed in the latest commit!

**Review by segeeslice — APPROVED** [review]
Thanks for the changes! I think that looks a lot better

**Pull request merged**